### PR TITLE
Added ability to listen on unix sockets

### PR DESCRIPTION
**Pull request by jcrugzz**
Implements #20. I have been working on the tests but I wanted to ask if there is a good way to test the unix socket case without creating more and more `unix.sock` files in the directory. First time dealing with them. @mmalecki thoughts?

### CODE REVIEW

File: lib/godot/net/client.js [hunk 1]
```diff
@@ -158,6 +173,11 @@ Client.prototype.connect = function (port, host, callback) {
       callback();
     }
   }
+  else if (this.type === 'unix') {
+    this.socket = new net.Socket({type: 'unix' });
+    this.socket.setEncoding('utf8');
```
Comment (blakmatrix) [inline lib/godot/net/client.js:178]: Is there a reason to force this encoding type? http://nodejs.org/api/stream.html#stream_stream_setencoding_encoding

Comment (blakmatrix) [inline lib/godot/net/client.js:178]: might be a good chance to utilize `options.encoding`

Comment (jcrugzz) [inline lib/godot/net/client.js:178]: hmm I suppose not. I was just following what was done for TCP.

**Comment by jcrugzz** [comment]
Thanks to a little help from @blakmatrix I got the tests working for the unix sockets.

**Comment by indexzero** [comment]
Cherry-picked. Thanks!

**Pull request closed without merging**